### PR TITLE
Fixes #89 and #91

### LINKED ISSUE #89: MAVEN_MIRROR_URL or similar mechanism not supported
There is no way to specify a maven mirror unless:

- a new assemble script is provided to support it
- a new settings.xml is provided

Should there be a way to support MAVEN MIRRORs like EAP image. If this is non standard, find a standard way for maven, but do provide something that can be Overriden in master's BuildDefaults envs.

### PR DESCRIPTION
**Pull request by jorgemoralespou**
* This PR allows to deploy .ear .war .jar and .rar
* Allows to provide own settings.xml through MAVEN_SETTINGS
* Allows to provide MAVEN_MIRROR_URL in default settings.xml in the images

### CODE REVIEW

File: 10.0/s2i/bin/assemble [hunk 1]
```diff
@@ -197,20 +276,14 @@ if [ -f "$LOCAL_SOURCE_DIR/pom.xml" ]; then
...
-  cp $LOCAL_SOURCE_DIR/target/*.war $DEPLOY_DIR >& /dev/null
-  chgrp -R 0 $DEPLOY_DIR
-  chmod -R g+rw $DEPLOY_DIR
+  copy_artifacts "target" war ear rar jar
 fi
 if [ -d $LOCAL_SOURCE_DIR/deployments ]; then
-  cp $LOCAL_SOURCE_DIR/deployments/*.war $DEPLOY_DIR >& /dev/null
-  chgrp -R 0 $DEPLOY_DIR
-  chmod -R g+rw $DEPLOY_DIR
+  copy_artifacts "deployments" war ear rar jar
```
Comment (bparees) [inline 10.0/s2i/bin/assemble]: all 3 of these still need chmod/chgrp (or copy_artifacts needs to do it)

Comment (jorgemoralespou) [inline 10.0/s2i/bin/assemble]: @bparees Why?
Permissions are fixed on the LOCAL_SOURCE_DIR (https://github.com/jorgemoralespou/sti-wildfly/blob/a5e07b92aaf1b3a77657705bf2a61839ba411dce/10.0/s2i/bin/assemble#L221-L223) and then artifacts are copied so permissions should be correct for deployments in source Git.

I'll add the fix, but I'm wondering if we're adding a lot of complexity to the assemble just to be sure things work without really validating that things are not working.

**Comment by jorgemoralespou** [comment]
Some fixes added. Default settings from EAP image.
Added back -Popenshift profile selection (although I don't like it) and better detection of settings.xml file in case MAVEN_SETTINGS env is provided.
chmod and chgrp added to copy_artifacts

File: 10.0/s2i/bin/assemble (hunk 1 above)
Comment (bparees) [inline 10.0/s2i/bin/assemble]: hm, perhaps you are correct.

as long as the end result is that when you oc rsh into the pod, you can delete/overwrite the files, that's sufficient.

**Comment by bparees** [comment]
lgtm, can you squash (and if your testing reveals the extra chmod/chgrp is indeed not needed, you can remove that too)

**Comment by jorgemoralespou** [comment]
Closed in favor of #93

**Pull request closed without merging**